epoxy-processor/src/main/java/com/airbnb/epoxy/EpoxyProcessor.java

```diff
@@ -662,8 +662,8 @@ private MethodSpec generateEquals(ClassToGenerateInfo helperClass) {
               .endControlFlow();
         }
       } else {
-        builder.beginControlFlow("if ($L != null && that.$L == null"
-                + " || $L == null && that.$L != null)",
+        builder.beginControlFlow("if (($L != null && that.$L == null)"
```

Comment:
I believe we could also simplify this to `if(($L == null) != (that.$L == null))` if you're so inclined